internal/operator/common.go

```diff
@@ -360,6 +371,82 @@ func initializeControllerWorkerCounts() {
 	}
 }
 
+// initializeOperatorBackrestSecret ensures the generic pgBackRest configuration
+// is available
+func initializeOperatorBackrestSecret(clientset kubernetes.Interface, namespace string) error {
+	var isNew, isModified bool
+
+	ctx := context.TODO()
+
+	// determine if the Secret already exists
+	secret, err := clientset.
+		CoreV1().Secrets(namespace).
+		Get(ctx, config.SecretOperatorBackrestRepoConfig, metav1.GetOptions{})
+
+	// if there is a true error, return. Otherwise, initialize a new Secret
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			return err
+		}
+
+		secret = &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: config.SecretOperatorBackrestRepoConfig,
+			},
+			Data: map[string][]byte{},
+		}
+		isNew = true
+	}
+
+	// if the default fields are all set, return
+	defaultKeysSet := true
+
+	for _, key := range defaultBackrestRepoConfigKeys {
+		defaultKeysSet = defaultKeysSet && len(secret.Data[key]) != 0
+	}
+
+	if defaultKeysSet {
+		return nil
+	}
+
+	// otherwise, set the defaults
```

Comment:
This logic is already covered by the following loop and `isModified`.
```suggestion
	// set any missing defaults
```

Reply:
Oh oops. Will adjust.